projects/vdk-core/src/vdk/internal/builtin_plugins/connection/decoration_cursor.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from typing import Container
+from typing import Optional
+from typing import Tuple
+
+from vdk.internal.builtin_plugins.connection.pep249.interfaces import PEP249Cursor
+
+
+class ManagedOperation:
+    """
+    Query and parameters DTO.
+    """
+
+    def __init__(self, operation: str, parameters: Optional[Container]):
+        self.__operation = operation
+        self.__parameters = parameters
+        self.__operation_decorated = operation
+        self.__parameters_decorated = parameters
+
+    def get_initial_operation(self):
+        return self.__operation
+
+    def get_initial_parameters(self):
+        return self.__parameters
+
+    def get_operation(self):
+        return self.__operation_decorated
+
+    def get_parameters(self):
+        return self.__parameters_decorated
+
+    def get_operation_parameters_tuple(self) -> Tuple[str, Optional[Container]]:
```

Comment:
Seems useful to me, for example:
```super().execute(*managed_operation.get_operation_parameters_tuple())```

Reply:
I think we should minimize our interface surface .

Interfaces take time to learn. A class with a huge interface is not likely to be used well and may well be off-putting in the first place. By keeping a small, focused set of methods, you make it easier for clients to find out what the class is and what it can do.

This focus is also important for the class designer/maintainers. A common problem with the class design is to make classes do too much. Focusing on the essentials helps keep cruft out of the class, allowing it to focus on doing one job and doing it well.

As a fan of Martin Flower, this is quoted from 
https://martinfowler.com/bliki/MinimalInterface.html (as I cannot speak so eloquently)

I also watched some time ago to a talk about shallow vs deep interfaces that I am going to try to find and send to you. This is a concept talked by John Ousterhout (A Philosophy of Software Design). But this article seems relevant: https://medium.com/@nakabonne/depth-of-module-f62dac3c2fdb